### PR TITLE
Group exceptions by status

### PR DESCRIPTION
**Pull request by Ewentualnie**

### CODE REVIEW

**Comment by Mo-og** [comment]
|File|Coverage [94.02%]|:green_apple:|
|:-|:-:|:-:|
|[ApplicationErrorException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FApplicationErrorException.java)|100%|:green_apple:|
|[NotFoundException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FNotFoundException.java)|100%|:green_apple:|
|[BadRequestException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FBadRequestException.java)|100%|:green_apple:|
|[PeriodValidator.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fvalidators%2FPeriodValidator.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[CandidateService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCandidateService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[CoordinatorService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCoordinatorService.java)|100%|:green_apple:|
|[BookingService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingService.java)|93.68%|:green_apple:|
|[InterviewerSlotValidator.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fvalidators%2FInterviewerSlotValidator.java)|81.48%|:green_apple:|
|[InterviewerService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/96978dcb636770e0f5c7201d4b44c4a8426c8721/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FInterviewerService.java)|65.99%|:green_apple:|

|Total Project Coverage|69.14%|:green_apple:|
|:-|:-:|:-:|

File: src/test/java/com/intellias/intellistart/interviewplanning/services/InterviewerServiceTest.java [hunk 1]
```diff
@@ -145,73 +94,24 @@ void testGetRelevantInterviewerSlotsForInvalidUser() {
     when(userRepository
         .existsById(-1L))
         .thenReturn(false);
-    assertThrows(InterviewerNotFoundException.class,
+    assertThrows(NotFoundException.class,
         () -> interviewerService.getRelevantInterviewerSlots(-1L));
   }
 
-  @Test
```
Comment (Mo-og) [inline src/test/java/com/intellias/intellistart/interviewplanning/services/InterviewerServiceTest.java:152]: Why removed?

**Review by Mo-og — CHANGES_REQUESTED** [review]
Last PR was 75% project coverage. Now it is 69%. Should be increased

**Comment by olesiayakovenko** [comment]
Why were the changes from the previous pull request removed?

File: src/test/java/com/intellias/intellistart/interviewplanning/services/InterviewerServiceTest.java (hunk 1 above)
Comment (Ewentualnie) [reply src/test/java/com/intellias/intellistart/interviewplanning/services/InterviewerServiceTest.java:152]: mistake while merge =(

**Comment by Ewentualnie** [comment]
> 

mistake while merge local branches =(
add removed code after meet

**Comment by Mo-og** [comment]
|File|Coverage [98.63%]|:green_apple:|
|:-|:-:|:-:|
|[ApplicationErrorException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FApplicationErrorException.java)|100%|:green_apple:|
|[NotFoundException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FNotFoundException.java)|100%|:green_apple:|
|[InvalidInputException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FInvalidInputException.java)|100%|:green_apple:|
|[PeriodValidator.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fvalidators%2FPeriodValidator.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[CandidateService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCandidateService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[CoordinatorService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCoordinatorService.java)|100%|:green_apple:|
|[InterviewerService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FInterviewerService.java)|97.96%|:green_apple:|
|[BookingService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingService.java)|93.68%|:green_apple:|
|[InterviewerSlotValidator.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/7e557021a9bdde17fc226185bcc15414c23af3a0/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fvalidators%2FInterviewerSlotValidator.java)|81.48%|:green_apple:|

|Total Project Coverage|73.26%|:green_apple:|
|:-|:-:|:-:|

**Comment by Mo-og** [comment]
|File|Coverage [98.67%]|:green_apple:|
|:-|:-:|:-:|
|[ApplicationErrorException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FApplicationErrorException.java)|100%|:green_apple:|
|[NotFoundException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FNotFoundException.java)|100%|:green_apple:|
|[InvalidInputException.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fexceptions%2FInvalidInputException.java)|100%|:green_apple:|
|[PeriodValidator.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fvalidators%2FPeriodValidator.java)|100%|:green_apple:|
|[UserService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FUserService.java)|100%|:green_apple:|
|[CandidateService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCandidateService.java)|100%|:green_apple:|
|[BookingLimitService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingLimitService.java)|100%|:green_apple:|
|[CoordinatorService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FCoordinatorService.java)|100%|:green_apple:|
|[InterviewerService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FInterviewerService.java)|97.96%|:green_apple:|
|[BookingService.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservices%2FBookingService.java)|93.68%|:green_apple:|
|[InterviewerSlotValidator.java](https://github.com/Mo-og/intellistart-java-2022-debuggers/blob/426893b9c089285a197a7fb53fd8a50cbf298a08/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fvalidators%2FInterviewerSlotValidator.java)|81.48%|:green_apple:|

|Total Project Coverage|77.01%|:green_apple:|
|:-|:-:|:-:|

File: src/test/java/com/intellias/intellistart/interviewplanning/exceptions/InvalidInputExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import com.intellias.intellistart.interviewplanning.exceptions.ApplicationErrorException.ErrorCode;
+import java.time.DayOfWeek;
+import org.junit.jupiter.api.Test;
+
+class InvalidInputExceptionTest {
+
+  private static InvalidInputException exception;
+
+  @Test
+  void invalidBoundariesTest() {
```
Comment (Mo-og) [inline src/test/java/com/intellias/intellistart/interviewplanning/exceptions/InvalidInputExceptionTest.java:14]: We should test that services (or validators etc.) throw exceptions when they are supposed to. Unit tests for exceptions themselves is too much, I think.

**Review by Mo-og — COMMENTED** [review]
Code is fine. @olesiayakovenko are all changes present?

**Review by Mo-og — APPROVED** [review]

**Review by olesiayakovenko — APPROVED** [review]

**Pull request merged**